test/hotspot/jtreg/compiler/c2/irTests/LShiftINodeIdealizationTests.java

```diff
@@ -71,4 +83,52 @@ public int test1(int x) {
     public int test2(int x) {
         return (x >>> 2022) << 2022;
     }
+
+    @Test
+    @IR(failOn = { IRNode.RSHIFT })
+    @IR(counts = { IRNode.AND, "1", IRNode.LSHIFT, "1" })
+    // Checks (x >> 4) << 8 => (x << 4) & -16
+    public int test3(int x) {
+        return (x >> 4) << 8;
+    }
+
+    @Test
+    @IR(failOn = { IRNode.URSHIFT })
+    @IR(counts = { IRNode.AND, "1", IRNode.LSHIFT, "1" })
+    // Checks (x >>> 4) << 8 => (x << 4) & -16
+    public int test4(int x) {
+        return (x >>> 4) << 8;
+    }
+
+    @Test
+    @IR(failOn = { IRNode.LSHIFT })
+    @IR(counts = { IRNode.AND, "1", IRNode.RSHIFT, "1" })
+    // Checks (x >> 8) << 4 => (x >> 4) & -16
+    public int test5(int x) {
+        return (x >> 8) << 4;
+    }
+
+    @Test
+    @IR(failOn = { IRNode.LSHIFT })
+    @IR(counts = { IRNode.AND, "1", IRNode.URSHIFT, "1" })
+    // Checks (x >>> 8) << 4 => (x >>> 4) & -16
+    public int test6(int x) {
+        return (x >>> 8) << 4;
+    }
+
+    @Test
+    @IR(failOn = { IRNode.RSHIFT })
+    @IR(counts = { IRNode.AND, "1", IRNode.LSHIFT, "1" })
+    // Checks (x >> 4) << 8 => (x << 4) & 0xFF00
```

Comment:
```suggestion
    // Checks ((x >> 4) & 0xFF) << 8 => (x << 4) & 0xFF00
```

Right? Same for test8 and the corresponding long tests.

Reply:
Oh, nice catch! It seems I had mistyped there.